packages/terra-dropdown-button/src/SplitButton.jsx

```diff
@@ -63,21 +74,50 @@ class SplitButton extends React.Component {
     super(props);
 
     this.handleDropdownButtonClick = this.handleDropdownButtonClick.bind(this);
+    this.handlePrimaryButtonClick = this.handlePrimaryButtonClick.bind(this);
     this.handleDropdownRequestClose = this.handleDropdownRequestClose.bind(this);
     this.handlePrimaryKeyDown = this.handlePrimaryKeyDown.bind(this);
     this.handlePrimaryKeyUp = this.handlePrimaryKeyUp.bind(this);
     this.handleCaretKeyDown = this.handleCaretKeyDown.bind(this);
     this.handleCaretKeyUp = this.handleCaretKeyUp.bind(this);
+    this.setButtonNode = this.setButtonNode.bind(this);
+    this.getButtonNode = this.getButtonNode.bind(this);
+    this.setListNode = this.setListNode.bind(this);
+
+    this.state = {
+      isOpen: false, caretIsActive: false, primaryIsActive: false, isKeyboardEvent: false,
+    };
+  }
+
+  setListNode(element) {
+    this.dropdownList = element;
+  }
+
+  setButtonNode(node) {
+    this.buttonNode = node;
+  }
 
-    this.state = { isOpen: false, caretIsActive: false, primaryIsActive: false };
+  getButtonNode() {
+    return this.buttonNode;
   }
 
   handleDropdownButtonClick() {
+    if (this.state.isOpen) {
+      this.setState({ isKeyboardEvent: false });
+    }
     this.setState(prevState => ({ isOpen: !prevState.isOpen }));
   }
 
+  handlePrimaryButtonClick(event) {
+    event.stopPropagation();
+    event.currentTarget.focus();
```

Comment:
This works but should we instead set `isPrimaryActive` to true in state? That seems to automatically take care of styles and that would remain consistent with how click keydown event handles the focus.

Reply:
we need this because setting primaryIsActive state will not set focus styles in safari and firefox so we need to set focus manually. 
reference : https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Button#Clicking_and_focus

Reply:
Why is `stopPropagation` necessary? I don't see this logic in `terra-button`. Also, I'd leave the MDN link in the code comment for code readers to understand in the future. 

https://github.com/cerner/terra-core/blob/ce4c1c129427a28161c0c079c8c9647e27c9d98f/packages/terra-button/src/Button.jsx#L137-L148

Reply:
I think here it was added with reference to [onClick() event of Item.jsx](https://github.com/cerner/terra-core/blob/3d4cbdca1a5c96ae211cdcf5bdb5847ecfdc965c/packages/terra-dropdown-button/src/Item.jsx#L51). Since onClick()  is on nested element `div` and not on `li` this could be added as safety check to prevent event from propagating further. 

I don't think we need it here and as well as in Item.jsx because I don't see any of the click() event defined on parent elements of Item and Primary Button. 

and I have also tested by removing this in all Browsers ( IE10 and 11 ) so I think for now we can remove this.